lib/kubernetes-deploy/errors.rb

```diff
@@ -8,6 +8,15 @@ def initialize(name, context)
       super("Namespace `#{name}` not found in context `#{context}`")
     end
   end
+
+  class DeploymentTimeoutError < FatalDeploymentError
+    attr_reader :resources
+    def initialize(resources)
+      @resources = resources
+      super("Resources #{@resources.map(&:name).join(',')} failed due to timeouts.")
```

Comment:
For consistency with the other messaging (e.g. right now we see `Successfully deployed 1 resource, failed to deploy 1 resource, and resources web failed due to timeouts.`), and because this could be a long list, I think this should be a count, not a list. 

Since we're trying to clarify that timeouts aren't exactly failures, might it be more helpful to not use "failed" in this message? e.g. "gave up watching X resources", "timed out waiting for X resources", "reached the watch timeout for X resources", I dunno.